txnsync/bloomFilter_test.go

```diff
@@ -290,32 +298,28 @@ func TestBloomFilterTest(t *testing.T) {
 
 	for _, filterFactory = range filters {
 
-		genesisHash := crypto.Hash([]byte("gh"))
-		genesisID := "gID"
-
 		var s syncState
 		s.node = &justRandomFakeNode{}
-		var encodingParams requestParams
-
-		for encodingParams.Modulator = 0; encodingParams.Modulator < 3; encodingParams.Modulator++ {
-			txnGroups := getTxnGroups(genesisHash, genesisID)
-			bf := s.makeBloomFilter(encodingParams, txnGroups, nil)
-			for _, tx := range txnGroups {
-				ans := bf.test(tx.GroupTransactionID)
-				if bf.encodingParams.Modulator == 0 {
-					require.False(t, ans)
-				} else if bf.encodingParams.Modulator <= 1 ||
-					txidToUint64(tx.GroupTransactionID)%uint64(bf.encodingParams.Modulator) ==
-						uint64(bf.encodingParams.Offset) {
-					require.True(t, ans)
-				} else {
-					require.False(t, ans)
-				}
-			}
+		var err error
+		txnGroups := getTxnGroups(testingGenesisHash, testingGenesisID)
+
+		filter, filterType := filterFactory(len(txnGroups), &s)
+		for _, txnGroup := range txnGroups {
+			filter.Set(txnGroup.GroupTransactionID[:])
+		}
+		var enc encodedBloomFilter
+		enc.BloomFilterType = byte(filterType)
+		enc.BloomFilter, err = filter.MarshalBinary()
+		require.NoError(t, err)
+
+		testableBf, err := decodeBloomFilter(enc)
+		require.NoError(t, err)
+		for _, tx := range txnGroups {
+			ans := testableBf.test(tx.GroupTransactionID)
```

Comment:
Why is the Modulator variations dropped here?
test checks the modulator value.

Reply:
good point. I'll add that back. ( i.e. I think that we don't need this for the encoding part.. but we want to verify the testing )